sdk/src/main/java/com/hedera/hashgraph/sdk/BatchTransaction.java

```
@@ -0,0 +1,157 @@
+        this.transactions.clear();
+        this.transactions.addAll(transactions);
+        return this;
+    }
+
+    /**
+     * Append a transaction to the list of transactions this BatchTransaction will execute.
+     *
+     * @param transaction The transaction to be added
+     * @return {@code this}
+     */
+    public BatchTransaction addTransaction(Transaction<?> transaction) {
+        Objects.requireNonNull(transaction);
+        requireNotFrozen();
+        //transaction.publicKeys.size()  >0
+        this.transactions.add(transaction);
+        return this;
+    }
+
+    /**
+     * Get the list of transactions this BatchTransaction is currently configured to execute.
+     *
+     * @return The list of transactions
+     */
+    public List<Transaction> getTransactions() {
+        return transactions;
+    }
+
+    /**
+     * Get the list of transaction IDs of each inner transaction of this BatchTransaction.
+     *
+     * **NOTE**: this will return undefined data until the transaction IDs for each inner
+     * transaction actually get generated/set (i.e. this BatchTransaction has been executed).
+     * This is provided to the user as a convenience feature in case they would like to get
+     * the receipts of each individual inner transaction.
+     *
+     * @return The list of inner transaction IDs
+     */
+    public List<TransactionId> getInnerTransactionIds() {
+        if (innerTransactionIds == null) {
+            return Collections.emptyList();
+        }
+        return Collections.unmodifiableList(innerTransactionIds);
+    }
+
+    @Override
+    void validateChecksums(Client client) throws BadEntityIdException {
+        for (Transaction<?> transaction : transactions) {
+            transaction.validateChecksums(client);
+        }
+    }
+
+    /**
+     * Initialize from the transaction body.
+     */
+    void initFromTransactionBody() {
+        var body = sourceTransactionBody;
+        System.out.println("hii");
+    }
```

It might be impossible to create batch txn from a protobuf body, since in the protobuf body we have a list of signed transaction bytes list, which I don't think could be converted to list of transactions.
We might need to throw `UnsupportedOperationException` here.
WDYT @rwalworth @gsstoykov @SimiHunjan ?